apps/token-e2e/src/integration/flow/staking-flow.cy.js

```diff
@@ -1,28 +1,44 @@
-const pageSpinner = 'splash-loader';
-const menuBar = 'nav';
-const validatorList = '[data-testid="node-list-item-name"]';
-const removeStakeRadioButton = '[data-testid="remove-stake-radio"]';
-const tokenAmountInputBox = '[data-testid="token-amount-input"]';
-const tokenSubmitButton = '[data-testid="token-input-submit-button"]';
+/// <reference types="cypress" />
+const stakeValidatorList = '[data-testid="node-list-item-name"]';
+const stakeValidatorWithinList = '[data-testid="node-list-item"]';
+const stakeRemoveStakeRadioButton = '[data-testid="remove-stake-radio"]';
+const stakeTokenAmountInputBox = '[data-testid="token-amount-input"]';
+const stakeTokenSubmitButton = '[data-testid="token-input-submit-button"]';
 const stakeNextEpochValue = '[data-testid="stake-next-epoch"]';
-const vegaWalletContainer = '[data-testid="vega-wallet"]';
-
+const stakeThisEpochValue = '[data-testid="stake-this-epoch"]';
+const stakeAddStakeRadioButton = '[data-testid="add-stake-radio"]';
+const stakeMaximumTokens = '[data-testid="token-amount-use-maximum"]';
 const vegaWalletPublicKeyShort = Cypress.env('vegaWalletPublicKeyShort');
+const vegaWalletAssociatedBalance = '[data-testid="currency-value"]';
+const vegaWalletUnstakedBalance =
+  '[data-testid="vega-wallet-balance-unstaked"]';
+const vegaWalletStakedBalances =
+  '[data-testid="vega-wallet-balance-staked-validators"]';
+const vegaWalletThisEpochBalances =
+  '[data-testid="vega-wallet-balance-this-epoch"]';
+const vegaWalletNextEpochBalances =
+  '[data-testid="vega-wallet-balance-next-epoch"]';
+const ethWalletAssociatedBalances =
+  '[data-testid="eth-wallet-associated-balances"]';
+const ethWalletTotalAssociatedBalance = '[data-testid="currency-locked"]';
+const ethWalletContainer = '[data-testid="ethereum-wallet"]';
+const txTimeout = { timeout: 40000 };
+const epochTimeout = { timeout: 10000 };
```

Comment:
Suggestion non-blocking: These timeouts are used in quite a few places and also environment dependant so I think they should be Cypress environment variables.

Reply:
Agreed